### PR TITLE
Unify Faraday config under local module instead of global prepend

### PR DESCRIPTION
**Pull request by gregorbg**
CC @FinnIckler

### CODE REVIEW

**Comment by FinnIckler** [comment]
looks good! Let me check out the branch and try

File: app/controllers/api/internal/v1/api_controller.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def validate_wca_token
     # Renew our token if it has expired or is close to expiring
     Vault.auth_token.renew_self if vault_token_data[:ttl] < 300
 
-    client = Faraday.new(url: EnvConfig.VAULT_ADDR)
+    client = Faraday.new(url: EnvConfig.VAULT_ADDR, &FaradayConfig)
```
Comment (FinnIckler) [inline app/controllers/api/internal/v1/api_controller.rb:16]: If you look into the file we actually parse our own requests here

**Review by FinnIckler — CHANGES_REQUESTED** [review]

File: app/controllers/api/internal/v1/api_controller.rb (hunk 1 above)
Comment (gregorbg) [reply app/controllers/api/internal/v1/api_controller.rb:16]: Good catch! Fixed :wink:

**Review by FinnIckler — APPROVED** [review]
LGTM

**Pull request merged**